brion/spikeReport.cpp

```diff
@@ -75,59 +77,138 @@ SpikeReport::~SpikeReport()
     delete _impl;
 }
 
-const URI& SpikeReport::getURI() const
-{
-    return _impl->plugin->getURI();
-}
 
-SpikeReport::ReadMode SpikeReport::getReadMode() const
+void SpikeReport::close()
 {
-    return _impl->plugin->getReadMode();
+    if( !_impl->plugin->isClosed() )
+    {
+        if( _impl->threadPool.hasPendingJobs() )
+        {
+            // interrupt the jobs
+            _impl->plugin->_interrupted = true;
+            // As only one thread in the pool,
+            // this blocks until the jobs arefinished
+            _impl->threadPool.post([]{}).get();
+        }
+        _impl->plugin->close();
+        _impl->plugin->setClosed();
+    }
 }
 
-float SpikeReport::getStartTime() const
+const URI& SpikeReport::getURI() const
 {
-    return _impl->plugin->getStartTime();
+    return _impl->plugin->getURI();
 }
 
-float SpikeReport::getEndTime() const
+float SpikeReport::getCurrentTime() const
 {
-    return _impl->plugin->getEndTime();
+    return _impl->plugin->getCurrentTime();
 }
 
-const Spikes& SpikeReport::getSpikes() const
+SpikeReport::State SpikeReport::getState() const
 {
-    return _impl->plugin->getSpikes();
+    return _impl->plugin->getState();
 }
 
-void SpikeReport::writeSpikes( const Spikes& spikes )
+bool SpikeReport::isClosed() const
 {
-    _impl->plugin->writeSpikes( spikes );
+    return _impl->plugin->isClosed();
 }
 
-bool SpikeReport::waitUntil( const float timeStamp, const uint32_t timeout )
+void SpikeReport::interrupt()
 {
-    return _impl->plugin->waitUntil( timeStamp, timeout );
+
+    if( _impl->threadPool.hasPendingJobs() )
+    {
+        _impl->plugin->_interrupted = true;
+        // As only one thread in the pool,
+        // this blocks until the jobs arefinished
+        _impl->threadPool.post([]{}).get();
+        _impl->plugin->_interrupted = false;
```

Comment:
I don't like the idea of implementing the interruption writing a member variable of the plugin directly from here, it should be a call to an interrupt method.